File: src/main/java/org/micoli/php/codeStyle/configuration/CodeStyle.java
package org.micoli.php.codeStyle.configuration;

import io.swagger.v3.oas.annotations.media.Schema;

public final class CodeStyle {
    @Schema(
            description = "Code style field property as in com.intellij.psi.codeStyle.CommonCodeStyleSettings",
            example = "ALIGN_MULTILINE_PARAMETERS_IN_CALLS")
    public String styleAttribute;

    @Schema(description = "a boolean value true/false or an int value", example = "false")
    public String value;
}

File: src/main/kotlin/org/micoli/php/codeStyle/configuration/CodeStyle.kt
package org.micoli.php.codeStyle.configuration

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonProperty
import io.swagger.v3.oas.annotations.media.Schema

class CodeStyle {
    @JsonProperty(required = true)
    @Schema(
        description =
            "Code style field property as in com.intellij.psi.codeStyle.CommonCodeStyleSettings",
        example = "ALIGN_MULTILINE_PARAMETERS_IN_CALLS")
    lateinit var styleAttribute: String

    @JsonProperty(required = true)
    @Schema(description = "a boolean value true/false or an int value", example = "false")
    lateinit var value: String

    @JsonIgnore
    fun isFullyInitialized(): Boolean {
        return ::styleAttribute.isInitialized && ::value.isInitialized
    }
}
